src/main/java/org/bptlab/cepta/DataToDatabase.java

```diff
@@ -0,0 +1,159 @@
+package org.bptlab.cepta;
+
+import com.github.jasync.sql.db.ConnectionPoolConfigurationBuilder;
+import com.github.jasync.sql.db.QueryResult;
+import com.github.jasync.sql.db.pool.ConnectionPool;
+import com.github.jasync.sql.db.postgresql.PostgreSQLConnection;
+import com.github.jasync.sql.db.postgresql.PostgreSQLConnectionBuilder;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import org.apache.flink.api.common.functions.MapFunction;
+
+public class DataToDatabase<Object> implements MapFunction<Object, Object> {
+  static final String HOST = "localhost";
+  static final String DATABASE = "dbs1_imdb";
+  static final String USER = "tester";
+  static final String PASSWORD = "password";
+  static final Integer PORT = 5432;
+  private String table_name;
+  public DataToDatabase(String table){
+    this.table_name = table;
+  }
+  @Override
+  public Object map(Object dataSet) throws Exception {
+    insert(dataSet);
+    return dataSet;
+  }
+  public void insert(Object dataSet)
+      throws NoSuchFieldException, IllegalAccessException {
+
+    // Connection to PostgreSQL DB
+    ConnectionPool<PostgreSQLConnection> connection;
+
+    ConnectionPoolConfigurationBuilder config = new ConnectionPoolConfigurationBuilder();
+    config.setUsername(USER);
+    config.setPassword(PASSWORD);
+    config.setHost(HOST);
+    config.setPort(PORT);
+    config.setDatabase(DATABASE);
+    config.setMaxActiveConnections(100);
+    connection = PostgreSQLConnectionBuilder.createConnectionPool(config);
+
+    // stores strings of values and columns for sql query
+    String[] colsAndVals = columnsAndValuesToString(dataSet);
+
+    // Create query
+    String query = "INSERT INTO " + table_name + colsAndVals[0]
+        + " VALUES " + colsAndVals[1] + ";";
+    System.out.println(query);
+
+    // send query
+    CompletableFuture<QueryResult> future = connection.sendPreparedStatement(query);
+
+    // really execute query and get query result
+    try{
+      QueryResult result = future.get();
+    }catch (InterruptedException | ExecutionException e){
+      System.out.println("Could not get result");
+    }
+
+    // Close the connection pool
+    try{
+      connection.disconnect().get();
+    }catch (ExecutionException | InterruptedException e){
+      System.out.println("Could not disconnect");
+    }
+  }
+
+  private String[] columnsAndValuesToString(Object dataSet)
```

Comment:
Naming of method feels off. the And smells a bit and it feels like it produces one string instead of an array of strings

Reply:
I think there should be two separate arrays one for values and one for column names or pairs in the one array.

Reply:
That is because we only add columns to the string that actually have a value and are not null. If we separate those methods we have nearly the same code and work twice.
So I can not think of a better way than having this one method.